tests/pykafka/test_protocol.py

```diff
@@ -256,5 +256,95 @@ def test_offset_fetch_response(self):
         self.assertEqual(response.topics[b'emmett.dummy'][0].offset, 1)
 
 
+class TestGroupMembershipAPI(unittest2.TestCase):
+    maxDiff = None
+
+    def test_consumer_group_protocol_metadata(self):
+        meta = protocol.ConsumerGroupProtocolMetadata()
+        msg = meta.get_bytes()
+        self.assertEqual(
+            msg,
+            bytearray(b'\x00\x00\x00\x00\x00\x01\x00\ndummytopic\x00\x00\x00\x0ctestuserdata')
```

Comment:
To make this more readable to humans, you could consider stuffing data into the docstring or another string which is converted into the bytearray, i.e. write a utility to convert this:

```python
"""
00 00 00 00 00 01 00
dummytopic
00 00 00 0c
testuserdata
```

```python
bytearray(b'\x00\x00\x00\x00\x00\x01\x00\ndummytopic\x00\x00\x00\x0ctestuserdata')
```

In other words, the utility trims whitespace and converts `\d+` to bytes. This kind of utility pays off quickly for writing tests when developing protocol drives.